src/ProjectSystemTools/RoslynLogging/RoslynWorkspaceStructureLogger.cs

```diff
@@ -159,26 +159,30 @@ public static void Log(IServiceProvider serviceProvider, string path)
 #pragma warning disable VSTHRD002 // Avoid problematic synchronous waits -- this is fine since it's a Roslyn API
                     var compilation = project.GetCompilationAsync(threadedWaitCallback.CancellationToken).Result;
 #pragma warning restore VSTHRD002 // Avoid problematic synchronous waits
-                    var compilationReferencesElement = new XElement("compilationReferences");
-                    projectElement.Add(compilationReferencesElement);
 
-                    foreach (var reference in compilation.References)
+                    if (compilation != null)
                     {
-                        compilationReferencesElement.Add(CreateElementForPortableExecutableReference(reference));
-                    }
+                        var compilationReferencesElement = new XElement("compilationReferences");
+                        projectElement.Add(compilationReferencesElement);
 
-                    projectElement.Add(CreateElementForCompilation(compilation));
+                        foreach (var reference in compilation.References)
+                        {
+                            compilationReferencesElement.Add(CreateElementForPortableExecutableReference(reference));
+                        }
```

Comment:
FYI the `XElement` constructor accepts enumerables, so you can write things like:

```c#
var compilationReferencesElement = new XElement(
    "compilationReferences",
    compilation.References.Select(CreateElementForPortableExecutableReference));
```

Also below, for diagnostics.

It also ignores any null values, which can be handy.

Sometime these allow constructing the entire document with a single expression.

Reply:
I didn't change any of that code, it just got indented 😛